srcts/src/bindings/input/selectInput.ts

```diff
@@ -142,6 +142,10 @@ class SelectInputBinding extends InputBinding {
         };
       };
 
+      if (hasDefinedProperty(data, "value")) {
+        // Requested new `selected` choices, so reset value before new options
+        selectize.clear();
+      }
       selectize.clearOptions();
```

Comment:
```suggestion
      selectize.clearOptions();
      // If a new `selected` value is provided, also clear the current selection (otherwise it gets added as an option).
      // Note: although the selectize docs suggest otherwise, as of selectize.js >v0.15.2, 
      // .clearOptions() no longer implicitly .clear()s (see #3967)
      if (hasDefinedProperty(data, "value")) {
        selectize.clear();
      }
```